include/rtc/global.hpp

```diff
@@ -34,6 +34,22 @@ RTC_CPP_EXPORT void InitLogger(LogLevel level, LogCallback callback = nullptr);
 RTC_CPP_EXPORT void Preload();
 RTC_CPP_EXPORT std::shared_future<void> Cleanup();
 
+struct UnhandledStunRequest {
+	std::string localUfrag;
+	std::string remoteUfrag;
+	std::string remoteHost;
+	uint16_t remotePort;
+};
+
+RTC_CPP_EXPORT typedef std::function<void(UnhandledStunRequest request, void* userPtr)> UnhandledStunRequestCallback;
```

Comment:
An `std::function` should not have a user pointer as it has its own context. The user can use bind, lambda capture, or a callable object.

Additionally, all callbacks in the library should be implemented with `synchronized_callback` to prevent race conditions between calling and setting/resetting.

Reply:
Fixed the user pointer part.

I'm not sure I follow about `synchronized_callback` - where it's used elsewhere here a callback has been passed in that's stored as a class member.  That member can get overwritten so I can see why we need to synchronize on it's access.

Here we're passing a reference to a function to `libjuice` and all state is passed along with the function invocation - do we still need `synchronized_callback`?

Reply:
You are right that it's not needed if you store only a pointer to an external function object, since the concern is pushed to the user. However, I don't think it's a good idea, as stated in https://github.com/paullouisageneau/libdatachannel/pull/1211#discussion_r1922347908.